Kernel/Devices/HID/USB/MouseDevice.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2022, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Try.h>
+#include <Kernel/Bus/USB/USBDescriptors.h>
+#include <Kernel/Bus/USB/USBRequest.h>
+#include <Kernel/Bus/USB/USBTransfer.h>
+#include <Kernel/Devices/DeviceManagement.h>
+#include <Kernel/Devices/HID/USB/MouseDevice.h>
+#include <Kernel/KString.h>
+
+namespace Kernel {
+
+ErrorOr<NonnullRefPtr<USBMouseDevice>> USBMouseDevice::try_create(USB::Device const& usb_device)
+{
+    auto device = TRY(DeviceManagement::try_create_device<USBMouseDevice>(usb_device));
+    TRY(device->create_and_start_polling_process());
+    return *device;
+}
+
+ErrorOr<void> USBMouseDevice::create_and_start_polling_process()
+{
+    auto kernel_process_name = TRY(KString::try_create("USB mouse handler"sv));
+    VERIFY(m_attached_usb_device->interrupt_pipe());
+    TRY(Process::create_kernel_process(move(kernel_process_name), [&]() {
```

Comment:
I'm a little rusty but if I'm remembering correctly instead of having to create an entirely new process here, you should be able to get away with just calling submit_interrupt_in_transfer once, since the UHCI driver will take care of the polling for you. And then in the destructor you would call cancel_async_transfer to do cleanup.

Reply:
Yeah right, I discovered this later this morning by another bug (and only if I looked at this comment before!).
This is fixed by now :)